test/src/main/java/io/strimzi/test/k8s/Exec.java

```diff
@@ -9,111 +9,285 @@
+    }
+
+    public int getRetCode() {
+        LOGGER.info("Process: {}", process);
+        if (isRunning()) {
+            return -1;
+        } else {
+            return process.exitValue();
+        }
+    }
+
+
+    /**
+     * Method executes external command
+     *
+     * @param command arguments for command
+     * @return execution results
+     */
+    public static ExecResult exec(String... command) {
+        return exec(Arrays.asList(command));
+    }
 
     /**
-     * Executes the given command in a subprocess.
-     * @throws KubeClusterException if the process returned a non-zero status code, or if anything else went wrong.
+     * Method executes external command
+     *
+     * @param command arguments for command
+     * @return execution results
      */
-    static ProcessResult exec(String... cmd) throws KubeClusterException {
-        return exec(Arrays.asList(cmd));
+    public static ExecResult exec(List<String> command) {
+        return exec(null, command, 0, false);
     }
 
     /**
-     * Executes the given command in a subprocess.
-     * @throws KubeClusterException if the process returned a non-zero status code, or if anything else went wrong.
+     * Method executes external command
+     *
+     * @param command arguments for command
+     * @return execution results
      */
-    static ProcessResult exec(List<String> cmd) throws KubeClusterException {
-        return exec(null, cmd);
+    public static ExecResult exec(String input, List<String> command) {
+        return exec(input, command, 0, false);
     }
-    static ProcessResult exec(String input, List<String> cmd) throws KubeClusterException {
-        File out = null;
-        File err = null;
+
+    /**
+     * Method executes external command
+     * @param command arguments for command
+     * @param timeout timeout for execution
+     * @param logToOutput log output or not
+     * @return execution results
+     */
+    public static ExecResult exec(String input, List<String> command, int timeout, boolean logToOutput) {
+        int ret = 1;
+        ExecResult execResult;
         try {
-            out = File.createTempFile(Exec.class.getName(), Integer.toString(cmd.hashCode()));
-            out.deleteOnExit();
-            err = File.createTempFile(Exec.class.getName(), Integer.toString(cmd.hashCode()));
-            err.deleteOnExit();
-
-            LOGGER.trace("{}", join(" ", cmd));
-            ProcessBuilder pb = new ProcessBuilder(cmd)
-                    .redirectOutput(out)
-                    .redirectError(err);
-            Process p = pb.start();
-            OutputStream outputStream = p.getOutputStream();
-            if (input != null) {
-                LOGGER.trace("With stdin {}", input);
-                outputStream.write(input.getBytes(Charset.defaultCharset()));
+            Exec executor = new Exec();
+            ret = executor.execute(input, command, timeout);
+            synchronized (LOCK) {
```

Comment:
Why is this necessary?

Reply:
It's for keep some order in logging in case we have multiple executor running on the same time